cmd/siklu/bank_management.c

```diff
@@ -69,4 +69,43 @@ struct software_bank_t* bank_management_get_current_bank(void) {
 	create_bank_management_info_for_bank(bank);
 	
 	return bank;
-}
\ No newline at end of file
+}
+
+struct software_bank_t* bank_management_switch_current_bank(struct software_bank_t *bank) {
+	u_char *fdt = NULL;
+	int8_t boot_tries_left = -1; // negative is disabled
+	const char *config_boot_tries_left;
+
+	fdt = siklu_read_fdt_from_mtd_part(CONFIG_SIKLU_BANK_MGMT_MTD_PART);
+	if (! fdt) {
+		printk(KERN_ERR "Could not read bank management info from \"%s\"\n",
+			   CONFIG_SIKLU_BANK_MGMT_MTD_PART);
+		return bank;
+	}
+
+	config_boot_tries_left = siklu_fdt_getprop_string(fdt, "/", PROP_BOOT_TRIES_LEFT, NULL);
+	if (IS_ERR(config_boot_tries_left)) {
+		printf("Could not read boot_tries_left\n");
+		return bank;
+	}
+
+	printf("boot_tries_left = %s\n", config_boot_tries_left);
+	boot_tries_left = simple_strtol(config_boot_tries_left, NULL, 10);
+	if (boot_tries_left < 0) {
+		return bank;
+	} else {
```

Comment:
maybe just remove the else?

Reply:
done

Reply:
also added additional check for wrong bank value